runtime-decompiler/src/main/java/org/fife/ui/hex/swing/HexSearch.java

```diff
@@ -0,0 +1,161 @@
+                    try {
+                        int i = Integer.parseInt(s, 16);
+                        arr.add(Integer.valueOf(i).byteValue());
+                    } catch (NumberFormatException e) {
+                        throw e;
+                    }
+                }
+                break;
+
+            default:
+                return null;
+        }
+        return arr;
+    }
+
+    public void searchHexCode(String str, HexSearchOptions type) {
+        try {
+            ArrayList<Byte> arr = getByteArray(str, type);
+            int byteCount = hex.getByteCount();
+            start = 0;
+            end = 0;
+            counter = 0;
+            found = false;
+            for (int i = 0; i < byteCount; i++) {
+                if (arr.get(0) == hex.getByte(i)) {
+                    if (checkIfMatches(arr, i)) {
+                        found = true;
+                        start = i;
+                        end = i + arr.size();
+                        break;
+                    }
+                }
+            }
+            if (found) {
+                hex.setSelectedRange(start, end - 1);
+            }
+        } catch (NumberFormatException e) {
+            System.err.println("NAN");
+        } catch (StringIndexOutOfBoundsException e) {
+            // Ignore
+        }
+    }
+
+    private boolean checkIfMatches(ArrayList<Byte> arr, int start) {
+        for (int i = 0; i < arr.size(); i++) {
+            if (!(arr.get(i) == hex.getByte(i + start))) {
+                return false;
+            }
+        }
+        return true;
+    }
+
+    public void next(String str, HexSearchOptions type) {
+        if (!found) {
+            return;
+        }
+        ArrayList<Byte> arr = getByteArray(str, type);
+        if (arr == null) {
+            return;
+        }
+        found = false;
+        int byteCount = hex.getByteCount();
+        for (int i = start + 1; i < byteCount; i++) {
+            if (arr.get(0) == hex.getByte(i)) {
+                if (checkIfMatches(arr, i)) {
+                    found = true;
+                    start = i;
+                    end = i + arr.size();
+                    break;
+                }
+            }
+        }
+        if (found) {
+            hex.setSelectedRange(start, end - 1);
+        } else {
+            found = true;
+        }
+    }
+
+    public void previous(String str, HexSearchOptions type) {
+        if (!found) {
+            return;
+        }
+        ArrayList<Byte> arr = getByteArray(str, type);
+        if (arr == null) {
+            return;
+        }
+        found = false;
+        int byteCount = hex.getByteCount();
+        for (int i = start - 1; i >= 0; i--) {
+            if (arr.get(0) == hex.getByte(i)) {
+                if (checkIfMatches(arr, i)) {
+                    found = true;
+                    start = i;
+                    end = i + arr.size();
+                    break;
```

Comment:
maybe replace start+end + wahtever by Class? Some SearchState? In that case, you can also properly return soemthing useful form this method